test/gtest/common/xpmem_test.h

```diff
@@ -20,33 +21,19 @@ extern "C" {
 
 #include <memory>
 
-template <typename T>
+template <typename T, int (*F)(T)>
 class handle {
+  T m_h;
+
  public:
-  typedef int (*dtor_t)(T);
-
-  handle() : m_h(0), m_dtor(empty_dtor) {}
-  handle(T h, dtor_t dtor) : m_h(h), m_dtor(dtor) {}
-  handle(handle &&other) { *this = std::move(other); }
-
-  handle &operator=(const handle&) = delete;
-  handle &operator=(handle &&other) {
-    m_h = std::move(other.m_h);
-    m_dtor = std::move(other.m_dtor);
-    other.m_h = 0;
-    other.m_dtor = empty_dtor;
-    return *this;
-  }
+  handle(T h) : m_h(h) {}
   operator T() const { return m_h; }
-
-  ~handle() { m_dtor(m_h); }
-
- private:
-  static int empty_dtor(T h) { return 0; }
-  T m_h;
-  dtor_t m_dtor;
+  ~handle() { F(m_h); }
 };
 
+using handle_segid = handle<xpmem_segid_t, xpmem_remove>;
```

Comment:
can we use std::unique_ptr here as well? and remove handle class altogether?

Reply:
fixed, but delete class is still needed to provide default constructor for segid and apid class members in test_multi_xpmem.